conda/core/subdir_data.py

```diff
@@ -355,25 +374,38 @@ def _load(self):
                 "304 NOT MODIFIED for '%s'. Updating mtime and loading from disk",
                 self.url_w_repodata_fn,
             )
-            touch(self.cache_path_json)
-            _internal_state = self._read_local_repodata(mod_etag_headers)
+            cache.refresh()
+            # touch(self.cache_path_json) # not anymore, or the .state.json is invalid
+            # self._save_state(mod_etag_headers)
+            _internal_state = self._read_local_repodata(cache.state)
             return _internal_state
         else:
-            if not isdir(dirname(self.cache_path_json)):
+            if not isdir(dirname(self.cache_path_json)):  # What happens if it is a directory?
                 mkdir_p(dirname(self.cache_path_json))
             try:
-                cache_path_json = self.cache_path_json
-                with io_open(cache_path_json, "w") as fh:
-                    fh.write(raw_repodata_str or "{}")
-                # quick thing to check for 'json matches stat', or store, check a message digest:
-                mod_etag_headers["mtime"] = pathlib.Path(cache_path_json).stat().st_mtime
-                self._save_state(mod_etag_headers)
+                if raw_repodata_str is RepodataOnDisk:
+                    # this is handled very similar to a 304. Can the cases be merged?
```

Comment:
We should eventually push more out to RepodataCache so that SubdirData only does package searching. Should be simpler and avoid some duplicate read/write/parse cycles

Reply:
In progress over here https://github.com/conda/conda/pull/12461

Reply:
Since the follow-on PR improves the RepodataOnDisk situation, we could have it return a str here always even when that sacrifices efficiency. Then this soon-to-be-eliminated version of the code would be a little simpler.